src/app/(tools)/square-image/square-tool.tsx

```diff
@@ -49,10 +49,71 @@ export const SquareTool: React.FC = () => {
     }
   };
 
+  const calculateAccentColor = (imgSrc: string): Promise<string> => {
+    return new Promise((resolve) => {
+      const img = new Image();
+      img.onload = () => {
+        const canvas = document.createElement("canvas");
+        const ctx = canvas.getContext("2d");
+        canvas.width = img.width;
+        canvas.height = img.height;
+
+        if (ctx) {
+          ctx.drawImage(img, 0, 0);
+          const imageData = ctx.getImageData(
+            0,
+            0,
+            canvas.width,
+            canvas.height,
+          ).data;
+          const colorMap = new Map<string, number>();
+
+          // Sample every 4th pixel to improve performance
+          for (let i = 0; i < imageData.length; i += 16) {
+            const r = imageData[i];
+            const g = imageData[i + 1];
+            const b = imageData[i + 2];
+
+            // Skip white and near-white colors
+            if (r !== undefined && g !== undefined && b !== undefined) {
+              if (r > 240 && g > 240 && b > 240) continue;
+
+              const color = `rgb(${r},${g},${b})`;
+              colorMap.set(color, (colorMap.get(color) || 0) + 1);
+            }
+          }
+
+          let maxCount = 0;
+          let dominantColor = "white"; // Default to white if no dominant color found
+
+          colorMap.forEach((count, color) => {
+            if (count > maxCount) {
+              maxCount = count;
+              dominantColor = color;
+            }
+          });
+          console.log(dominantColor);
+          resolve(dominantColor);
+        } else {
+          resolve("white");
+        }
+      };
+      img.src = imgSrc;
+    });
+  };
+
   useEffect(() => {
     if (imageFile) {
       const reader = new FileReader();
-      reader.onload = () => {
+      reader.onload = async () => {
+        const imgSrc = reader.result as string;
+
+        // Calculate accent color first
+        if (backgroundColor === "accent") {
```

Comment:
imo this shouldn't be part of the reader.onload code, see if there's somewhere that it fits better after the upload refactor

Reply:
> imo this shouldn't be part of the reader.onload code, see if there's somewhere that it fits better after the upload refactor

done